### PR TITLE
[Fix] Monitoring and Provisioner payload parsing.

### PR DESCRIPTION
**Pull request by paul-amonson**
[Fix] Fabric adapter bug fixes.
[Feat] Docker containers now use fabric providers.

Signed-off-by: Paul Amonson <paul.d.amonson@intel.com>

### CODE REVIEW

**Comment by paul-amonson** [comment]
So the Travis CI reporting failed. The build actually passed:

https://travis-ci.org/github/unifiedcontrolsystem/dai-ds/builds/703946793

**Comment by paul-amonson** [comment]
Something in inventory is now failing...debugging now.

**Comment by paul-amonson** [comment]
Ok, due to reduced count of total branches, the inventory branch coverage dropped from 61% to 60%. Altered the build.gradle file appropriately.

File: config-files/FabricCritTelemetryProvider.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "connectAddress": "sms01-nmn.local",
```
Comment (yhuan47) [inline config-files/FabricCritTelemetryProvider.json:2]: Looks like the telemetry server host name is very consistent as "api-gw-service-nmn.local", could you update the default configuration to be aligned?

Comment (paul-amonson) [reply config-files/FabricCritTelemetryProvider.json:2]: All occurrences fixed. Will need to re-approve.

File: monitoring/src/test/groovy/com/intel/dai/monitoring/AdapterMonitoringNetworkBaseSpec.groovy [hunk 2]
```diff
@@ -1,100 +1,48 @@
...
+        when: underTest_.getConfigStream("unknownFile.json")
+        then: thrown(FileNotFoundException)
     }
 
-    def "Test Execute"() {
-        underTest_.callSuper = true
-        expect: underTest_.execute(Mock(NetworkListenerCore))
+    def "Test execute"() {
+        def core = Mock(NetworkListenerCore)
+        core.run() >>> [0, 1]
```
Comment (cmcheung-UCS) [inline monitoring/src/test/groovy/com/intel/dai/monitoring/AdapterMonitoringNetworkBaseSpec.groovy:39]: Is >>> a typo?  I am not familiar with this.

File: monitoring/src/test/groovy/com/intel/dai/monitoring/EnvironmentalProviderForeignBusSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+              ]
+            }
+          }
+        ]
+      }
+    ]
+  }
+}
+"""
+    def badJson_ = """
```
Comment (cmcheung-UCS) [inline monitoring/src/test/groovy/com/intel/dai/monitoring/EnvironmentalProviderForeignBusSpec.groovy:261]: May be a good idea to comment on why this is bad.

File: monitoring/src/test/groovy/com/intel/dai/monitoring/ProviderMonitoringNetworkForeignBusSpec.groovy [hunk 4]
```diff
@@ -1,38 +1,18 @@
...
-        when: ProviderMonitoringNetworkForeignBus.main(bad)
-        then: thrown(RuntimeException)
-    }
-
-    def "Test Main with no config file"() {
-        ProviderMonitoringNetworkForeignBus.main(args_)
+    def "Test ctor"() {
+        new ProviderMonitoringNetworkForeignBus(Mock(Logger), Mock(DataStoreFactory), Mock(AdapterInformation),
+                "./build/tmp/bmf.txt", 100L)
         expect: true
```
Comment (cmcheung-UCS) [inline monitoring/src/test/groovy/com/intel/dai/monitoring/ProviderMonitoringNetworkForeignBusSpec.groovy:16]: Maybe an expectation of no exception thrown is better.  Even better is to test that the object is well-formed.

File: monitoring/src/test/groovy/com/intel/dai/monitoring/AdapterMonitoringNetworkBaseSpec.groovy (hunk 2 above)
Comment (paul-amonson) [reply monitoring/src/test/groovy/com/intel/dai/monitoring/AdapterMonitoringNetworkBaseSpec.groovy:39]: No, it defines an array of in order return values. First time return 0, second time return 1.

File: monitoring/src/test/groovy/com/intel/dai/monitoring/ProviderMonitoringNetworkForeignBusSpec.groovy (hunk 4 above)
Comment (paul-amonson) [reply monitoring/src/test/groovy/com/intel/dai/monitoring/ProviderMonitoringNetworkForeignBusSpec.groovy:16]: I will address this in the next review. Yes, I do agree with you and its a good idea!

File: monitoring/src/test/groovy/com/intel/dai/monitoring/EnvironmentalProviderForeignBusSpec.groovy (hunk 3 above)
Comment (paul-amonson) [reply monitoring/src/test/groovy/com/intel/dai/monitoring/EnvironmentalProviderForeignBusSpec.groovy:261]: I will address this in the next review.

**Pull request merged**